src/ui/router/ApplicationRouter.js

```diff
@@ -185,16 +193,39 @@ class ApplicationRouter extends EventEmitter {
         this.set(path, searchParams.toString());
     }
 
+    hashChaged(p) {
+        this.emit('change:hash', p);
+        this.handleLocationChange(p);
+    }
+
+    changeHash(hash) {
+        location.hash = '#' + hash.replace(/#/g, '');
+    }
+
+    updateTimeSettings() {
+        const hash = `${this.currentLocation.path}?${this.currentLocation.getQueryString()}`;
+
+        this.changeHash(hash);
+    }
+
+    navigateToObject(objectLink) {
```

Comment:
Would navigate(hash) or something similar make more sense? Since we're passing in a hash/url and not a direct object? It was a little confusing to me where it's being called and passing in differently named variables.